### PR TITLE
Add support for capabilities options in services

### PR DESCRIPTION
**Pull request by olljanat**
**- What I did**
Add support for capabilities options in services

Third step to address moby/moby#25885 , moby/moby#24862 and docker/swarmkit#1030

**- How I did it**
- Increased API version 1.41
- Added Capabilities option
- Bump swarmkit

**- How to verify it**
Create service with command:
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities":["CAP_NET_RAW","CAP_SYS_CHROOT"]}}}' \
  http://localhost/v1.41/services/create
```
Check log with command: `docker logs <container id>` which should print capabilities from inside of container.

Update capabilities
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities":["CAP_MKNOD"]}}}}' \
  http://localhost/v1.41/services/test/update?version=7139
```

Reset to default capabilities:
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities": null}}}}' \
  http://localhost/v1.41/services/test/update?version=7157
```

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

**Comment by olljanat** [comment]
Derek add label: rebuild/janky

**Comment by olljanat** [comment]
This is ready for review. @thaJeztah PTAL

File: vendor/github.com/docker/swarmkit/vendor.conf
Comment (thaJeztah) [inline vendor/github.com/docker/swarmkit/vendor.conf]: We probably want this one bumped as well (or merge https://github.com/moby/moby/pull/39269 first)

File: api/swagger.yaml
Comment (thaJeztah) [inline api/swagger.yaml]: Can you add `items` and `items -> type` here? Perhaps add an example value while we're at it;

```suggestion
              A list of kernel capabilities to be available for container (this overrides the default set).
            items:
              type: "string"
            example:
              - "CAP_NET_RAW"
              - "CAP_SYS_ADMIN"
              - "CAP_SYS_CHROOT"
              - "CAP_SYSLOG"
```

File: api/swagger.yaml
Comment (thaJeztah) [inline api/swagger.yaml]: Might be better to include this as a line in the description (perhaps change UNIX to Linux, or change to, e.g., "This option is not used by Windows containers")

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks! Left some comments inline 🤗

File: api/swagger.yaml
Comment (olljanat) [reply api/swagger.yaml]: Changed

**Comment by olljanat** [comment]
@thaJeztah rebased and updated based on your comments. Janky should pass after #39277 is merged.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by DigitalTargit** [comment]
How would you use this from a compose-file that you pushed at a swarm using "docker stack deploy" ?

**Comment by thaJeztah** [comment]
Compose file changes are not yet implemented, but you'd specify the list of capabilities that a service should get

**Comment by olljanat** [comment]
Yea. That part probably needs still some discussion and actual implementation after that. For example if we will actually support "--privileged" switch like requested on docker/swarmkit#1030 and #24862 or will we up force user to specify exact list of capabilities? (it will be still possible to specify all the capabilities so it more about how we want to implement user interface).

**Comment by DigitalTargit** [comment]
uggg this question has no business here ... but how do I get this build for testing on Ubuntu 16.04 ...

**Comment by thaJeztah** [comment]
You can compile a static binary with `make binary`, which builds the daemon and stores it in `bundles`. You can stop your current daemon and replace it (but definitely only recommended for testing)

Otherwise wait for this to be merged, and it should appear as a nightly build on download.docker.com

**Comment by DigitalTargit** [comment]
Cool ... I cloned https://github.com/olljanat/moby.git ... and it built. Since you approved the change does this mean this change gets merged tonight and show up in tonights nightly build ?

**Comment by thaJeztah** [comment]
You make me blush! All credits for this one really go to @olljanat!

We generally want two maintainers to review and give their LGTM before we merge; to be sure that a reviewer didn't miss anything (we're human 😁).

That said; anyone can help (and is more than welcome) to review PRs (more eyes never hurt)

This PR is fairly straightforward, so once another maintainer finds time to review, I expect it to be merged soon

**Comment by DigitalTargit** [comment]
I'm trying to test this build and ... uggg not sure what's wrong but the capabilities not showing up.

> dockerd -v                                                                                                                                                                          Docker version dev, build f787b23

> docker service ls
ID                  NAME                MODE                REPLICAS            IMAGE                        PORTS                                                                                                           10d5rmbuuz03        mongo_mongodb       replicated          1/1                 mongo:3.6.5

> curl --unix-socket /var/run/docker.sock   --header "Content-Type: application/json"   --request POST   --data '{"Name":"mongo_mongodb","TaskTemplate":{"ContainerSpec":{"Image":"mo ngo:3.6.5","Capabilities":["CAP_MKNOD"]}}}}'   http://localhost/v1.41/services/mongo_mongodb/update?version=28183                                                                                                           
{"Warnings":null}

> curl --unix-socket /var/run/docker.sock  http://localhost/v1.41/services/10d5rmbuuz03                                                                                               {"ID":"10d5rmbuuz035j2lk4gaafbbr","Version":{"Index":28184},"CreatedAt":"2019-05-29T15:26:29.220758109Z","UpdatedAt":"2019-05-29T23:49:41.198440323Z","Spec":{"Name":"mongo_mongodb","Labels":{},"TaskTemplate":{"ContainerSpec":{"Image":"mongo:3.6.5","Isolation":"default"},"ForceUpdate":0,"Runtime":"container"},"Mode":{"Replicated":{"Replicas":1}}},"PreviousSpec":{"Name":"mongo_mongodb","Labels":{},"TaskTemplate":{"ContainerSpec":{"Image": "mongo:3.6.5","Isolation":"default"},"ForceUpdate":0,"Runtime":"container"},"Mode":{"Replicated":{"Replicas":1}}},"Endpoint":{"Spec":{}}}

**Comment by DigitalTargit** [comment]
FYI ... I tried it out. I couldn't use it for what I was hoping to solve (using mongo with cifs remote storage) and honestly driver support for remote Azure using swarm is a huge mystery to me ... Azure just added NFS today). BUT I did test it this PR (created a swarm with one service) and I do see the capabilities returned in the JSON :)

**Review by kolyshkin — APPROVED** [review]
LGTM

**Pull request merged**
